src/Validator.php

```diff
@@ -76,16 +81,28 @@ public function validate($data): Result
 						if (is_bool($node)) {
 							$isValid = true;
 							break;
+						} elseif ($autoCast && ($node === '1' || $node === '0')) {
+							$node = (bool) $node;
+							$isValid = true;
+							break;
 						}
 					} elseif ($type === 'int') {
 						if (is_int($node)) {
 							$isValid = $this->validateNumber($node, $schema, $path, $errors);
 							break;
+						} elseif ($autoCast && is_string($node) && preg_match('#^-?[1-9][0-9]*+\z#', $node) === 1) {
+							$node = (int) $node;
+							$isValid = $this->validateNumber($node, $schema, $path, $errors);
+							break;
 						}
 					} elseif ($type === 'float') {
 						if (is_float($node)) {
 							$isValid = $this->validateNumber($node, $schema, $path, $errors);
 							break;
+						} elseif ($autoCast && is_string($node) && preg_match('#^-?[1-9][0-9]*+(\.[0-9]+)?\z#', $node) === 1) {
```

Comment:
This also wouldn't accept `0` and there are also other float notations (1e12, etc.), but I'm not sure if we want to support those. Here `is_numeric()` would probably do the job.